src/d3a/models/strategy/load_hours.py

```diff
@@ -244,7 +244,9 @@ def event_market_cycle(self):
             if self.hrs_per_day[current_day] <= FLOATING_POINT_TOLERANCE:
                 self.energy_requirement_Wh[market.time_slot] = 0.0
                 self.state.desired_energy_Wh[market.time_slot] = 0.0
-
+                for trade in market.trades:
+                    if trade.buyer_origin == self.owner.name:
+                        self.state.desired_energy_Wh[market.time_slot] += trade.offer.energy * 1000
```

Comment:
You are increasing the desired energy when a trade had happened? Shouldn't this value be decreased then?

Reply:
Removing it, as it no more needed for total_energy_demanded.